transport/src/main/java/io/netty/channel/ChannelOutboundBuffer.java

```diff
@@ -194,8 +194,7 @@ private void decrementPendingOutboundBytes(long size, boolean invokeLater, boole
         }
 
         long newWriteBufferSize = TOTAL_PENDING_SIZE_UPDATER.addAndGet(this, -size);
-        if (notifyWritability && (newWriteBufferSize == 0
-            || newWriteBufferSize <= channel.config().getWriteBufferLowWaterMark())) {
+        if (notifyWritability && newWriteBufferSize < channel.config().getWriteBufferLowWaterMark()) {
```

Comment:
I see. Does it matter though? For one, `newWriteBufferSize == 0` is only true on the last call of `remove()` before the buffer is empty. So in practice, you reduce "volatile reads" only for a tiny fraction of the invocations. Additionally, `newWriteBufferSize == 0` is not correct, as `0 < 0` is false, so according to the spec the channel should not be writable.

Also, having `newWriteBufferSize == 0` makes using `ChannelOutboundBuffer` for HTTP/2 child channels impractical, as whenever the buffer is empty, the channel would become writable, even if there is no flow control window left.

Thoughts?